woocommerce/woocommerce-android
# Shipping Labels Onboarding: Installation UI

Part of: #6587


### Description
This PR adds the installation UI, with the animations for the transition from the pre-install step, the changes required some refactorings, as I'm learning about Compose animation APIs, and I found out that the usage of `animateEnterExit` made the code a bit cleaner.

The additional change is replacing the `cancel` button with a cross-icon as this was confirmed by the design team.

NB: the installation itself is not implemented yet.

### Testing instructions
1. Make sure your store is in US, and uses USD as the currency, but doesn't have WCShip.
2. Create an order eligible for shipping label creation (ie: not virtual/downloadable and not eligible for IPP).
3. Open the order in the app.
4. Click on the "Get WooCommerce Shipping" button.
5. Click on "Add extension" to the store.
6. Click on "Proceed with installation".
7. Confirm the animation and transition look and work as expected.

You can also run the preview `PreviewInstallWCShippingScreen` if you want to jump directly into the screen.

### Images/gif
https://user-images.githubusercontent.com/1657201/176007989-b83a3ab1-736b-42f5-bb42-f3ae69d4a426.mov
